pkg/config/validation/validation.go

```diff
@@ -3477,3 +3479,53 @@ func validateTelemetryProviders(providers []*telemetry.ProviderRef) error {
 	}
 	return nil
 }
+
+// ValidateWasmPlugin validates a WasmPlugin.
+var ValidateWasmPlugin = registerValidateFunc("ValidateWasmPlugin",
+	func(cfg config.Config) (Warning, error) {
+		spec, ok := cfg.Spec.(*extensions.WasmPlugin)
+		if !ok {
+			return nil, fmt.Errorf("cannot cast to service entry")
+		}
+
+		errs := Validation{}
+		errs = appendValidation(errs,
+			validateWorkloadSelector(spec.Selector),
+			validateWasmPluginURL(spec.Url),
+			validateWasmPluginSHA(spec),
+		)
+		return errs.Unwrap()
+	})
+
+func validateWasmPluginURL(pluginURL string) error {
+	if pluginURL == "" {
+		return fmt.Errorf("url field needs to be set")
+	}
+	validSchemes := map[string]bool{
+		"": true, "file": true, "http": true, "https": true, "oci": true,
+	}
+
+	u, err := url.Parse(pluginURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse url: %s", err)
+	}
+	if _, found := validSchemes[u.Scheme]; !found {
+		return fmt.Errorf("url contains unsupported scheme: %s", u.Scheme)
+	}
+	return nil
+}
+
+func validateWasmPluginSHA(plugin *extensions.WasmPlugin) error {
```

Comment:
What happens when sha verification fails?
We are not enforcing it here because the validator does not do call outs.
How does the user know? 
@bianpengyuan  cc

Reply:
User will need to look at the metrics (https://istio.io/latest/docs/ops/configuration/extensibility/wasm-module-distribution/#monitor-wasm-module-distribution see `istio_agent_wasm_remote_fetch_count`), or agent/istiod logs.

Reply:
yeah the code is here:
http: https://github.com/istio/istio/blob/master/pkg/wasm/cache.go#L140
oci: https://github.com/istio/istio/blob/master/pkg/wasm/cache.go#L151